packages/braintree-integration/src/braintree-integration-service.ts

```diff
@@ -182,6 +195,10 @@ export default class BraintreeIntegrationService {
         }
     }
 
+    getSessionId(): Promise<string | undefined> {
+        return this.getDataCollector().then(({ deviceData }) => deviceData);
+    }
```

Comment:
It might be better to use async await here:
```suggestion
    async getSessionId(): Promise<string | undefined> {
        const { deviceData } = await this.getDataCollector();
        
        return deviceData;
    }
```